examples/src/main/java/ai/djl/examples/training/transferlearning/TrainAmazonReviewRanking.java

```diff
@@ -144,24 +144,21 @@ private static CsvDataset getDataset(
                 .build();
     }
 
-    private static Block addFreezeLayer(Predictor<NDList, NDList> embedder) {
+    private static Block addFreezeLayer(Block embedder) {
         if ("PyTorch".equals(Engine.getDefaultEngineName())) {
             return new LambdaBlock(
                     ndList -> {
                         NDArray data = ndList.singletonOrThrow();
-                        try {
-                            return embedder.predict(
-                                    new NDList(
-                                            data.toType(DataType.INT64, false),
-                                            data.getManager()
-                                                    .full(data.getShape(), 1, DataType.INT64),
-                                            data.getManager()
-                                                    .arange(data.getShape().get(1))
-                                                    .toType(DataType.INT64, false)
-                                                    .broadcast(data.getShape())));
-                        } catch (TranslateException e) {
-                            throw new IllegalArgumentException("embedding error", e);
-                        }
+                        return embedder.forward(
+                                new ParameterStore(),
```

Comment:
Can we avoid calling forward manually?

Reply:
Yes ideally I want to avoid this as well - I am wondering whether we can use the TranslatorOptions mechanism to convert the dataset output to the compatible model input? Haven't looked into it much yet.

Agreed that we should avoid calling forward... I'll investigate some options.